File: src/main/java/com/esprit/models/activiteevent.java
package com.esprit.models;

import java.util.Objects;

public class activiteevent {
    private int id;
    private String horaire;
    private int nbrparticipant;
    private int idEvent;
    private int idTypeActivite ;
    public activiteevent(int id, String horaire, int nbrparticipant, int idEvent,int idTypeActivite ) {
        this.id = id;
        this.horaire = horaire;
        this.nbrparticipant = nbrparticipant;
        this.idEvent = idEvent;
        this.idTypeActivite = idTypeActivite;
    }
    public activiteevent( String horaire, int nbrparticipant, int idEvent ,int idTypeActivite ) {
        this.horaire = horaire;
        this.nbrparticipant = nbrparticipant;
        this.idEvent = idEvent;
        this.idTypeActivite = idTypeActivite;
    }

    public activiteevent() {
        this.horaire = horaire;
        this.nbrparticipant = nbrparticipant;
    }

    public int getIdTypeActivite() {
        return idTypeActivite;
    }

    public void setIdTypeActivite(int idTypeActivite) {
        this.idTypeActivite = idTypeActivite;
    }

    public int getId() {
        return id;
    }

    public int getIdEvent() {
        return idEvent;
    }
    public void setIdEvent(int idEvent) {
        this.idEvent = idEvent;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getHoraire() {
        return horaire;
    }

    public void setHoraire(String horaire) {
        this.horaire = horaire;
    }

    public int getNbrparticipant() {
        return nbrparticipant;
    }

    public void setNbrparticipant(int nbrparticipant) {
        this.nbrparticipant = nbrparticipant;
    }


    @Override
    public String toString() {
        return "activiteevent{" +
                "id=" + id +
                ", horaire='" + horaire + '\'' +
                ", nbrparticipant=" + nbrparticipant +
                ", idEvent=" + idEvent +
                ", idTypeActivite=" + idTypeActivite +
                '}';
    }


    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        activiteevent that = (activiteevent) o;
        return id == that.id && nbrparticipant == that.nbrparticipant && idEvent == that.idEvent && idTypeActivite == that.idTypeActivite && Objects.equals(horaire, that.horaire);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, horaire, nbrparticipant, idEvent, idTypeActivite);
    }
}

File: src/Entity/Activiteevent.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\ActiviteeventRepository;

#[ORM\Entity(repositoryClass: ActiviteeventRepository::class)]
#[ORM\Table(name: 'activiteevent')]
class Activiteevent
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): self
    {
        $this->id = $id;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $horaire = null;

    public function getHoraire(): ?string
    {
        return $this->horaire;
    }

    public function setHoraire(string $horaire): self
    {
        $this->horaire = $horaire;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $nbrparticipant = null;

    public function getNbrparticipant(): ?int
    {
        return $this->nbrparticipant;
    }

    public function setNbrparticipant(int $nbrparticipant): self
    {
        $this->nbrparticipant = $nbrparticipant;
        return $this;
    }

    #[ORM\ManyToOne(targetEntity: Event::class, inversedBy: 'activiteevents')]
    #[ORM\JoinColumn(name: 'idEvent', referencedColumnName: 'id')]
    private ?Event $event = null;

    public function getEvent(): ?Event
    {
        return $this->event;
    }

    public function setEvent(?Event $event): self
    {
        $this->event = $event;
        return $this;
    }

    #[ORM\ManyToOne(targetEntity: Typeactivite::class, inversedBy: 'activiteevents')]
    #[ORM\JoinColumn(name: 'idTypeActivite', referencedColumnName: 'id')]
    private ?Typeactivite $typeactivite = null;

    public function getTypeactivite(): ?Typeactivite
    {
        return $this->typeactivite;
    }

    public function setTypeactivite(?Typeactivite $typeactivite): self
    {
        $this->typeactivite = $typeactivite;
        return $this;
    }
    public function __toString(): string
    {
        $output = 'Activité #'.$this->getId();
        
        if (!empty($this->horaire)) {
            $output .= ' - '.$this->horaire;
        }
        
        if ($this->nbrparticipant > 0) {  // Correction: nbrparticipant au lieu de nbpparticipant
            $output .= ' ('.$this->nbrparticipant.' participants)';
        }
        
        // Utilisation des objets relationnels plutôt que des IDs directs
        if ($this->event !== null) {
            $output .= ' - Evt:'.$this->event->getId();
        }
        
        if ($this->typeactivite !== null) {
            $output .= ' - Type:'.$this->typeactivite->getId();
        }
        
        return $output;
    }
}
